utilities/models_setup.py

```diff
@@ -59,4 +64,6 @@ def generate_model(model_type: str, input_dim: int, dataset: [Data], config: dic
             num_conv_layers=config["num_conv_layers"],
         ).to(device)
 
-    return model
+    ddp_model = DDP(model, device_ids=None, output_device=None)
```

Comment:
device_ids will have to match with the device id that you use in the line above.